File: code/smv-debugger/src/main/java/org/fbme/smvDebugger/panel/condition/Conjunction.java
package org.fbme.smvDebugger.panel.condition;

import java.util.Map;

public class Conjunction extends BinaryExpression {
  public Conjunction(final Expression first, final Expression second) {
    super(first, second);
  }

  @Override
  public boolean evaluate(final Map<String, String> itemSimpleNameToValueMap) {
    return first.evaluate(itemSimpleNameToValueMap) && second.evaluate(itemSimpleNameToValueMap);
  }

  @Override
  public String getSymbol() {
    return "&&";
  }
}

File: code/smv-debugger/src/main/kotlin/org/fbme/smvDebugger/panel/condition/Conjunction.kt
package org.fbme.smvDebugger.panel.condition

class Conjunction(first: Expression, second: Expression) : BinaryExpression(first, second) {
    override fun evaluate(itemSimpleNameToValueMap: Map<String, String>): Boolean {
        return first.evaluate(itemSimpleNameToValueMap) && second.evaluate(itemSimpleNameToValueMap)
    }

    override val symbol: String
        get() = "&&"
}
